Geometry_Engine/Modify/RoundCoordinates.cs

```diff
@@ -0,0 +1,215 @@
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Base;
+using BH.oM.Geometry;
+
+namespace BH.Engine.Geometry
+{
+    public static partial class Modify
+    {
+        /***************************************************/
+        /**** Interface Methods                         ****/
+        /***************************************************/
+
+        [Description("Modifies a BHoM IGeometry's Points' coordinates to be rounded to the number of provided decimal places.")]
+        [Input("geometry", "The BHoM IGeometry to modify.")]
+        [Input("decimalPlaces", "The number of decimal places to round to, default 6.")]
+        [Output("geometry", "The modified BHoM IGeometry.")]
+        public static IGeometry IRoundCoordinates(this IGeometry geometry, int decimalPlaces = 6)
+        {
+            return RoundCoordinates(geometry as dynamic, decimalPlaces);
+        }
+
+        /***************************************************/
+
+        [Description("Modifies a BHoM ICurve's Points coordinates to be rounded to the number of provided decimal places.")]
+        [Input("curve", "The BHoM ICurve to modify.")]
+        [Input("decimalPlaces", "The number of decimal places to round to, default 6.")]
+        [Output("curve", "The modified BHoM ICurve.")]
+        public static ICurve IRoundCoordinates(this ICurve curve, int decimalPlaces = 6)
+        {
+            return RoundCoordinates(curve as dynamic, decimalPlaces);
+        }
+
+
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Modifies a BHoM Geometry Point to be rounded to the number of provided decimal places.")]
+        [Input("point", "The BHoM Geometry Point to modify.")]
+        [Input("decimalPlaces", "The number of decimal places to round to, default 6.")]
+        [Output("point", "The modified BHoM Geometry Point.")]
+        public static Point RoundCoordinates(this Point point, int decimalPlaces = 6)
+        {
+            return new Point
+            {
+                X = Math.Round(point.X, decimalPlaces),
+                Y = Math.Round(point.Y, decimalPlaces),
+                Z = Math.Round(point.Z, decimalPlaces),
+            };
+        }
+
+        /***************************************************/
+
+        [Description("Modifies a BHoM Geometry Line's end points to be rounded to the number of provided decimal places.")]
```

Comment:
Given a BHoM Line has a definition for an `End` point I wonder if we should change this description to remove the word `end` in the description to avoid confusion given the method is editing both start and end points.